src/socket/dhcpv4.rs

```diff
@@ -247,6 +251,18 @@ impl<'a> Socket<'a> {
         self.ignore_naks = ignore_naks;
     }
 
+    /// Set the server/client port
+    ///
+    /// Allows you to specify the ports used by DHCP.
+    /// This is meant to support esoteric usecases allowed by the dhclient program.
+    pub fn set_ports(&mut self, ports: (u16, u16)) {
+        self.ports = ports;
+    }
+
+    pub fn get_ports(&self) -> (u16, u16) {
```

Comment:
Required to perform the test in `ipv4.rs`. I can also make the field `pub` if a getter is not what we'd like.